sql/core/src/test/scala/org/apache/spark/sql/InjectRuntimeFilterSuite.scala

```diff
@@ -267,14 +267,20 @@ class InjectRuntimeFilterSuite extends QueryTest with SQLTestUtils with SharedSp
   // `MergeScalarSubqueries` can duplicate subqueries in the optimized plan, but the subqueries will
   // be reused in the physical plan.
   def getNumBloomFilters(plan: LogicalPlan, scalarSubqueryCTEMultiplicator: Int = 1): Integer = {
-    val numBloomFilterAggs = plan.collectWithSubqueries {
-      case Aggregate(_, aggregateExpressions, _) =>
-        aggregateExpressions.collect {
-          case Alias(AggregateExpression(bfAgg: BloomFilterAggregate, _, _, _, _), _) =>
-            assert(bfAgg.estimatedNumItemsExpression.isInstanceOf[Literal])
-            assert(bfAgg.numBitsExpression.isInstanceOf[Literal])
-            1
+    val numBloomFilterAggs = plan.collect {
+      case Filter(condition, _) => condition.collect {
+        case subquery: org.apache.spark.sql.catalyst.expressions.ScalarSubquery
```

Comment:
I like the previous code more as it precisely matches `Filter` and `ScalarSubquery`. +1 to this change.

My major concern is `scalarSubqueryCTEMultiplicator`. This makes the test really hard to write as we need to manually think about if scalar subqueries merging can be applied or not to the testing query.

Can we turn off this optimization completely in this test suite? We can add one more test to verify the case that scalar subqueries merging is beneficial to bloom filter join, by explicitly enabling the optimizer rule.

Reply:
https://github.com/apache/spark/pull/36361/commits/7ac7981a3f9ded78744f76ce1e2b6f6564aae7bf removes `scalarSubqueryCTEMultiplicator`, https://github.com/apache/spark/pull/36361/commits/9b1347eafa9537ad216af95b2d321948f0bcd1b0 adds a new test